eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -423,21 +424,26 @@ object Helpers {
       }
     }
 
-    def firstClosingFee(commitments: Commitments, localScriptPubkey: ByteVector, remoteScriptPubkey: ByteVector)(implicit log: LoggingAdapter): Satoshi = {
+    def firstClosingFee(commitments: Commitments, localScriptPubkey: ByteVector, remoteScriptPubkey: ByteVector, feeratePerKw: Long)(implicit log: LoggingAdapter): Satoshi = {
       import commitments._
       // this is just to estimate the weight, it depends on size of the pubkey scripts
       val dummyClosingTx = Transactions.makeClosingTx(commitInput, localScriptPubkey, remoteScriptPubkey, localParams.isFunder, Satoshi(0), Satoshi(0), localCommit.spec)
       val closingWeight = Transaction.weight(Transactions.addSigs(dummyClosingTx, dummyPublicKey, remoteParams.fundingPubKey, Transactions.PlaceHolderSig, Transactions.PlaceHolderSig).tx)
-      // no need to use a very high fee here, so we target 6 blocks; also, we "MUST set fee_satoshis less than or equal to the base fee of the final commitment transaction"
-      val feeratePerKw = Math.min(Globals.feeratesPerKw.get.blocks_6, commitments.localCommit.spec.feeratePerKw)
       log.info(s"using feeratePerKw=$feeratePerKw for initial closing tx")
       Transactions.weight2fee(feeratePerKw, closingWeight)
     }
 
+    def firstClosingFee(commitments: Commitments, localScriptPubkey: ByteVector, remoteScriptPubkey: ByteVector, feeEstimator: FeeEstimator, feeTargets: FeeTargets)(implicit log: LoggingAdapter): Satoshi = {
+      // we "MUST set fee_satoshis less than or equal to the base fee of the final commitment transaction"
+      val blockTarget = Math.max(feeTargets.mutualCloseBlockTarget, feeTargets.commitmentBlockTarget)
```

Comment:
We need to compare feerates, not block targets.

Reply:
We are comparing feerates, this code is the result of adapting:
```
// no need to use a very high fee here, so we target 6 blocks; also, we "MUST set fee_satoshis less than or equal to the base fee of the final commitment transaction"
val feeratePerKw = Math.min(Globals.feeratesPerKw.get.blocks_6, commitments.localCommit.spec.feeratePerKw)
```
into
```
// we "MUST set fee_satoshis less than or equal to the base fee of the final commitment transaction"
val blockTarget = Math.max(feeTargets.mutualCloseBlockTarget, feeTargets.commitmentBlockTarget)
val feeratePerKw = feeEstimator.getFeeratePerKw(blockTarget)
```
It's just a different way to extract the feerate, note that the `Math.min` of the 2 feerates has become `Math.max` of the confirmation target because the higher the target the lower the feerate.

Reply:
I understand what you did, but it doesn't work: there is no guarantee that the resulting fee will be lesser than the one used in the last commitment, because fees may vary widely. We *must* compare to `commitments.localCommit.spec.feeratePerKw`.

Reply:
That's correct the resulting feerate must be compared with the one used in the commitment transaction and it's not the case currently.

Reply:
In 419bf60 i used `Math.min(feeEstimator.getFeeratePerKw(blockTarget), commitments.localCommit.spec.feeratePerKw)` to do the proper check.

Reply:
Right, but the line above with the max on block targets is misleading. This convey the intent better:

```scala
val requestedFeerate = feeEstimator.getFeeratePerKw(feeTargets.mutualCloseBlockTarget)
// we "MUST set fee_satoshis less than or equal to the base fee of the final commitment transaction"
val feeratePerKw = Math.min(requestedFeerate, commitments.localCommit.spec.feeratePerKw)
```

Reply:
Sounds good to me, but note that with this change if `mutualCloseBlockTarget < commitmentBlockTarget` we are (likely) to make a spec violation, I propose to have a check done in the `NodeParam` to avoid that.

Reply:
@pm47 Nevermind my comment above, the suggested change is correct.